### PR TITLE
ZEN-15723: reporting events as displayed.

### PR DESCRIPTION
**Pull request by byshovets**
Updates for https://jira.zenoss.com/browse/ZEN-15723:
reporting only integer value displayed at infrastructure page.

### CODE REVIEW

File: Products/ZenUI3/browser/device_export.py [hunk 1]
```diff
@@ -13,10 +13,24 @@
...
 from Products.Zuul.facades.devicefacade import DeviceFacade
 from xml.etree.ElementTree import Element, tostring
+from Products.ZenEvents.EventClass import EventClass
 
 log = logging.getLogger('zen.deviceexporter')
 
 
+def event(values):
+    # Get event's types with non-empty count field
+    values = {x:y.get('count') for x,y in values.items() if y.get('count')}
```
Comment (jpeacock-zenoss) [inline Products/ZenUI3/browser/device_export.py]: The `if` clause will be false if `y.get('count')` returns zero.  Is this desired?  If not, you'll need to check explicitly for `None`.

File: Products/ZenUI3/browser/device_export.py [hunk 2]
```diff
@@ -13,10 +13,24 @@
 from Products.ZenUtils.jsonutils import unjson
 from Products.Zuul.facades.devicefacade import DeviceFacade
 from xml.etree.ElementTree import Element, tostring
+from Products.ZenEvents.EventClass import EventClass
 
 log = logging.getLogger('zen.deviceexporter')
 
 
+def event(values):
```
Comment (jpeacock-zenoss) [inline Products/ZenUI3/browser/device_export.py]: Please add a doc comment explaining the purpose of this function.

File: Products/ZenUI3/browser/device_export.py [hunk 3]
```diff
@@ -82,18 +85,7 @@ def csv(response, devices, fields):
...
-                            data.append(str(value.get(event_type).get('count')))
-                            data.append(str(value.get(event_type).get('acknowledged_count')))
-                    else:
-                        value = value.get('name', None)
-                if not field == 'events':
-                    data.append(str(value).strip() if value or value is 0 else '')
-            if not wrote_header:
-                writer.writerow(fields_to_write)
-                wrote_header = True
+                    value = event(value) if field == 'events' else value.get('name', None)
```
Comment (jpeacock-zenoss) [inline Products/ZenUI3/browser/device_export.py]: In the code `value.get('name', None)`, the `None` portion is redundant. The `get` method defaults to returning `None` if the dict does have the field.

File: Products/ZenUI3/browser/device_export.py [hunk 4]
```diff
@@ -82,18 +85,7 @@ def csv(response, devices, fields):
...
+                data.append(str(value).strip() if value or value is 0 else '')
```
Comment (jpeacock-zenoss) [inline Products/ZenUI3/browser/device_export.py]: When there are multiple checks in an `if` expression, put parenthesis around the test, e.g.
`str(value).strip() if (value or value is 0) else ''`.
This improves readability.

File: Products/ZenUI3/browser/device_export.py (hunk 1 above)
Comment (byshovets) [inline Products/ZenUI3/browser/device_export.py]: Yes, I do not need values where "count" is 0. Will change "non-empty" to "non-zero" in comment.

**Pull request merged**